tests/js/server/aql/aql-optimizer-rule-late-document-materialization-rocksdb.js

```diff
@@ -0,0 +1,362 @@
+/// limitations under the License.
+///
+/// Copyright holder is ArangoDB GmbH, Cologne, Germany
+///
+/// @author Yuriy Popov
+/// @author Copyright 2019, ArangoDB GmbH, Cologne, Germany
+////////////////////////////////////////////////////////////////////////////////
+
+let jsunity = require("jsunity");
+let db = require("@arangodb").db;
+let isCluster = require("internal").isCluster();
+
+function lateDocumentMaterializationRuleTestSuite () {
+  const ruleName = "late-document-materialization";
+  const numOfCollections = 4;
+  let collectionNames = [];
+  var i;
+  for (i = 0; i < numOfCollections; ++i) {
+    collectionNames.push("UnitTestsCollection" + i);
+  }
+  return {
+    setUpAll : function () {
+      for (i = 0; i < numOfCollections; ++i) {
+        db._drop(collectionNames[i]);
+      }
+
+      var collections = [];
+      for (i = 0; i < numOfCollections; ++i) {
+        collections.push(db._create(collectionNames[i], { numberOfShards: 3 }));
+      }
+
+      for (i = 0; i < numOfCollections; ++i) {
+        let type;
+        let fields;
+        if (i % 2 === 0) {
+          fields = ["obj.a", "obj.b", "obj.c"];
+        } else {
+          fields = ["tags.hop[*].foo.fo", "tags.hop[*].bar.br", "tags.hop[*].baz.bz"];
+        }
+
+        if (i < numOfCollections / 2) {
+          type = "hash";
+        } else {
+          type = "skiplist";
+        }
+
+        collections[i].ensureIndex({type: type, fields: fields});
+      }
+
+      for (i = 0; i < numOfCollections; ++i) {
+        if (i % 2 === 0) {
+          collections[i].save({ _key: 'c0',  "obj": {"a": "a_val", "b": "b_val", "c": "c_val", "d": "d_val"}});
+          collections[i].save({ _key: 'c1',  "obj": {"a": "a_val_1", "b": "b_val_1", "c": "c_val_1", "d": "d_val_1"}});
+          collections[i].save({ _key: 'c2',  "obj": {"a": "a_val", "b": "b_val_2", "c": "c_val_2", "d": "d_val_2"}});
+          collections[i].save({ _key: 'c3',  "obj": {"a": "a_val_3", "b": "b_val_3", "c": "c_val_3", "d": "d_val_3"}});
+        } else {
+          collections[i].save({ _key: 'c1_0',  "tags": {"hop": [{"foo": {"fo": "a_val_1"}}, {"bar": {"br": "bar_val"}}, {"baz": {"bz": "baz_val"}}]}});
+          collections[i].save({ _key: 'c1_1',  "tags": {"hop": [{"foo": {"fo": "foo_val_1"}}, {"bar": {"br": "bar_val_1"}}, {"baz": {"bz": "baz_val_1"}}]}});
+          collections[i].save({ _key: 'c1_2',  "tags": {"hop": [{"foo": {"fo": "a_val_1"}}, {"bar": {"br": "bar_val_2"}}, {"baz": {"bz": "baz_val_2"}}]}});
+          collections[i].save({ _key: 'c1_3',  "tags": {"hop": [{"foo": {"fo": "foo_val_3"}}, {"bar": {"br": "bar_val_3"}}, {"baz": {"bz": "baz_val_3"}}]}});
+        }
+      }
+    },
+
+    tearDownAll : function () {
+      for (i = 0; i < numOfCollections; ++i) {
+        try { db._drop(collectionNames[i]); } catch(e) {}
+      }
+    },
+    testNotAppliedDueToNoFilter() {
+      for (i = 0; i < numOfCollections; ++i) {
+        if (i % 2 === 0) {
```

Comment:
this is only testing collections with `i % 2 === 0`, collections with `i % 2 !== 0` don't seem to be tested at all.